lib/class-wp-rest-annotations-controller.php

```diff
@@ -0,0 +1,558 @@
+<?php
+/**
+ * Annotations REST API: WP_REST_Annotations_Controller class
+ *
+ * @package gutenberg
+ * @since [version]
+ */
+
+/**
+ * Controller which provides a REST API endpoint for Gutenberg to read, create and edit annotations.
+ * Annotations are stored as posts with a custom post type. They can be children of all other post types.
+ *
+ * @since [version]
+ *
+ * @see WP_REST_Controller
+ * @see WP_REST_Posts_Controller
+ */
+class WP_REST_Annotations_Controller extends WP_REST_Posts_Controller {
+	/**
+	 * Prepared query vars.
+	 *
+	 * @since [version]
+	 * @access protected
+	 *
+	 * @var array
+	 */
+	protected $prepared_query_vars = array();
+
+	/**
+	 * Constructs the controller.
+	 *
+	 * @since [version]
+	 * @access public
+	 */
+	public function __construct() {
+		parent::__construct( WP_Annotation_Utils::$post_type );
+
+		// `rest_base` is already configured via register_post_type().
+		$this->namespace = 'gutenberg/v1'; // @codingStandardsIgnoreLine - PHPCS false positive on 'namespace'.
```

Comment:
I feel it might be best to specify this string in some global config/variable instead, and reuse it from there i.e. if "_gutenberg/v1_" needs to change over time, then it can easily be overlooked if it's sitting inside some class. Not sure if this is a convention though.

Reply:
No question. Thanks for spotting that.

Reply:
Took another look at this. It does appear to be a convention followed by WP core and the reusable blocks controller also. I suppose it makes sense to leave it in the class definition too, since the class needs to be kept to support the previous version — should it change.

Reply:
Ok - got it. Thanks for letting me know :-)